src/internal/connector/graph/api/api.go

```diff
@@ -27,3 +27,30 @@ func NextLink(pl PageLinker) string {
 func NextAndDeltaLink(pl DeltaPageLinker) (string, string) {
 	return NextLink(pl), ptr.Val(pl.GetOdataDeltaLink())
 }
+
+type Valuer[T any] interface {
+	GetValue() []T
+}
+
+type PageLinkValuer[T any] interface {
+	PageLinker
+	Valuer[T]
+}
+
+// emptyDeltaLinker is used to convert PageLinker to DeltaPageLinker
+type emptyDeltaLinker[T any] struct {
+	PageLinkValuer[T]
+}
```

Comment:
Not much point in an exported constructor here.  Just export the struct and let callers use it to wrap PLVs

```suggestion
type EmptyDeltaLinker[T any] struct {
    PageLinkValuer[T]
}
```